### PR TITLE
Core: VS 16.5.0 Warning fixes

### PR DESCRIPTION
**Pull request by petersteneteg**
Also a workaround for #251

### CODE REVIEW

File: src/core/util/stringconversion.cpp [hunk 1]
```diff
@@ -43,27 +44,57 @@
...
-}
 #else
-std::wstring toWstring(const std::string& str) {
     auto state = std::mbstate_t();
-    const char* s = str.c_str();
-    size_t len = std::mbsrtowcs(nullptr, &s, 0, &state) + 1;
+    size_t len = std::mbsrtowcs(nullptr, &str.data(), 0, &state) + 1;
     std::wstring result(len, 0);
-    std::mbsrtowcs(&result[0], &s, result.size(), &state);
+    std::mbsrtowcs(result.data(), &str.data(), result.size(), &state);
```
Comment (r-englund) [inline src/core/util/stringconversion.cpp]: `mbsrtowcs` takes `const char **` so you need to pass in `&&str.data()` here and the line above. This is also true for `wcsrtombs`  
https://en.cppreference.com/w/cpp/string/multibyte/mbsrtowcs
https://en.cppreference.com/w/cpp/string/multibyte/wcsrtombs

**Review by r-englund — APPROVED** [review]

File: src/core/util/stringconversion.cpp (hunk 1 above)
Comment (petersteneteg) [reply src/core/util/stringconversion.cpp]: that's not the issue, problem is str.data() returns a const char* r-value which I can not take the adress of.. need to assign it to an l-value first...

File: modules/qtwidgets/include/modules/qtwidgets/inviwoqtutils.h [hunk 2]
```diff
@@ -225,6 +225,19 @@ IVW_MODULE_QTWIDGETS_API QSize emToPx(const QWidget* w, QSizeF);
 IVW_MODULE_QTWIDGETS_API int emToPx(const QWidget* w, double em);
 IVW_MODULE_QTWIDGETS_API int emToPx(const QFontMetrics& m, double em);
 
+
+
+/**
+ * Workaround for closing Widget using the "X" button in the titlebar
+ * This filter intercepts the CloseEvent and ignores it, and then hides the widget
+ */
+struct IVW_MODULE_QTWIDGETS_API WidgetCloseEventFilter : QObject {
```
Comment (ResearchDaniel) [inline modules/qtwidgets/include/modules/qtwidgets/inviwoqtutils.h:232]: Would call this HideOnCloseEventFilter or HideInsteadOfCloseEventFilter or similar

Comment (petersteneteg) [reply modules/qtwidgets/include/modules/qtwidgets/inviwoqtutils.h:232]: At least I added a comment ;)

**Pull request merged**